File: pkg/codegen/testing/test/testdata/logical-name-pp/nodejs/logical-name.ts
import * as pulumi from "@pulumi/pulumi";
import * as random from "@pulumi/random";

export = async () => {
    const config = new pulumi.Config();
    const configLexicalName = config.require("cC-Charlie_charlie.😃⁉️");
    const resourceLexicalName = new random.RandomPet("aA-Alpha_alpha.🤯⁉️", {prefix: configLexicalName});
    const outputLexicalName = resourceLexicalName.id;
    return {
        "bB-Beta_beta.💜⁉": outputLexicalName,
    };
}

File: pkg/codegen/testing/test/testdata/logical-name-pp/dotnet/logical-name.cs
using System.Collections.Generic;
using Pulumi;
using Random = Pulumi.Random;

await Deployment.RunAsync(() => 
{
    var resourceLexicalName = new Random.RandomPet("aA-Alpha_alpha.🤯⁉️");

    return new Dictionary<string, object?>
    {
        ["bB-Beta_beta.💜⁉"] = resourceLexicalName.Id,
    };
});

